R/TunerIrace.R

```diff
@@ -0,0 +1,94 @@
+#' * `eachTest` (`integer(1)`)
+#' * `confidence` (`numeric(1)`)
+#' * `capping` (`integer(1)`)
+#' * `cappingType` (`character(1)`)
+#' * `boundType` (`character(1)`)
+#' * `boundMax` (`numeric(1)`)
+#' * `boundDigits` (`integer(1)`)
+#' * `boundPar` (`numeric(1)`)
+#' * `boundAsTimeout` (`numeric(1)`)
+#'
+#' `show.irace.output` suppresses the output from [irace::irace()] and only prints output from this package.
+#' For the meaning of all other parameters, see [irace::defaultScenario()].
+#' Note that we have removed all control parameters which refer to the termination of the algorithm and
+#' where our terminators allow to obtain the same behavior.
+#'
+#' @templateVar id irace
+#' @template section_dictionary_tuners
+#'
+#' @family Tuner
+#' @export
+#' @examples
+#' # see ?Tuner
+TunerIrace = R6Class("TunerIrace", inherit = Tuner,
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function() {
+      ps = ParamSet$new(list(
+        ParamLgl$new("show.irace.output", default = FALSE),
+        ParamInt$new("debugLevel", default = 0, lower = 0),
+        ParamInt$new("seed"),
+        ParamDbl$new("postselection",default=0,lower=0,upper=1),
+        ParamInt$new("elitist",default=1,lower=0,upper=1),
+        ParamInt$new("elitistLimit",default=2,lower=0),
+        ParamInt$new("mu",default=5,lower=1),
+        ParamInt$new("softRestart",default=1,lower=0,upper=1),
+        ParamDbl$new("softRestartThreshold"),
+        ParamInt$new("digits", default = 4, lower = 0),
+        ParamFct$new("testType", default = "F-test", levels = c("F-test","t-test","t-test-bonferroni","t-test-holm")),
+        ParamInt$new("firstTest", default = 5, lower = 0),
+        ParamInt$new("eachTest",default = 1, lower = 1),
+        ParamDbl$new("confidence",default=0.95,lower=0,upper=1),
+        ParamInt$new("capping",default = 0,lower=0,upper=1),
+        ParamFct$new("cappingType",default="median",levels=c("median","mean","best","worst")),
+        ParamFct$new("boundType",default="candidate",levels=c("candidate","instance")),
+        ParamDbl$new("boundMax",default=0),
+        ParamInt$new("boundDigits",default=0),
+        ParamDbl$new("boundPar",default=1),
+        ParamDbl$new("boundAsTimeout",default=1)
+      ))
+
+      ps$values$show.irace.output = FALSE
+
+      super$initialize(
+        param_set = ps,
+        param_classes = c("ParamDbl","ParamInt","ParamFct","ParamLgl"),
+        properties = character(),
+        packages = "irace"
+      )
+    }
+  ),
+
+  private = list(
+    .tune = function(instance) {
+      ps = self$param_set$values[!(names(self$param_set$values) %in% c("show.irace.output"))]
+      g = if (self$param_set$values$show.irace.output) identity else capture.output # copied from mlr
+      g(irace::irace(scenario = c(makeScenario(instance), ps),
+                     parameters = paradoxToIrace(instance$param_set)))
+    }
+  )
```

Comment:
you should probably overload the `assign_result` function and ask the irace result on what configuration it believes to be best. See my mattermost comments on that.

Reply:
I'm trying to do this but I'm running into a problem where even though `irace` is correctly calling each configuration with new parameters and evaluating these on each instance, for some reason the archived instance parameters do not align with `irace`. Which would be fine except `irace` only returns the optimal parameters but not the final performance cost

Reply:
Is this resolved? Looks like it. Please mark as solved then.